parity-util-mem/src/impls.rs

```diff
@@ -36,31 +34,29 @@ malloc_size_of_is_0!(std::time::Duration);
 
 malloc_size_of_is_0!(U64, U128, U256, U512, H32, H64, H128, H160, H256, H264, H512, H520, Bloom);
 
-macro_rules! impl_elastic_array {
-	($name: ident, $dummy: ident, $size: expr) => {
-		impl<T> MallocSizeOf for $name<T>
+macro_rules! impl_smallvec {
+	($size: expr) => {
+		impl<T> MallocSizeOf for SmallVec<[T; $size]>
 		where
 			T: MallocSizeOf,
 		{
 			fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
-				self[..].size_of(ops)
+				let mut n = if self.spilled() {
+					self.capacity() * core::mem::size_of::<T>()
+				} else {
+					0
+				};
+				for elem in self.iter() {
+					n += elem.size_of(ops);
+				}
+				n
 			}
 		}
 	};
 }
 
-impl_elastic_array!(ElasticArray2, ElasticArray2Dummy, 2);
-impl_elastic_array!(ElasticArray4, ElasticArray4Dummy, 4);
-impl_elastic_array!(ElasticArray8, ElasticArray8Dummy, 8);
-impl_elastic_array!(ElasticArray16, ElasticArray16Dummy, 16);
-impl_elastic_array!(ElasticArray32, ElasticArray32Dummy, 32);
-impl_elastic_array!(ElasticArray36, ElasticArray36Dummy, 36);
-impl_elastic_array!(ElasticArray64, ElasticArray64Dummy, 64);
-impl_elastic_array!(ElasticArray128, ElasticArray128Dummy, 128);
-impl_elastic_array!(ElasticArray256, ElasticArray256Dummy, 256);
-impl_elastic_array!(ElasticArray512, ElasticArray512Dummy, 512);
-impl_elastic_array!(ElasticArray1024, ElasticArray1024Dummy, 1024);
-impl_elastic_array!(ElasticArray2048, ElasticArray2048Dummy, 2048);
```

Comment:
not sure about these removals, due to orphans rules this is the only place we can add these impls, so maybe it's ok to have them even if they're unused now?

Reply:
Hmm. Not a fan of dead code. They’re easy to add back should we ever need them yeah?

Reply:
yeah, but it's easy to miss impls we're not aware of, also it's a common practice add an impl for every reasonable size that can be used in the future, but I'm fine with leaving this as is